src/Java.Interop.Tools.Generator/Metadata/NamespaceTransform.cs

```diff
@@ -0,0 +1,114 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Xml.Linq;
+using Xamarin.Android.Tools;
+
+namespace Java.Interop.Tools.Generator
+{
+	public class NamespaceTransform
+	{
+		public string OldValue { get; }
+		public string NewValue { get; }
+		public bool IsStartsWith { get; }
+		public bool IsEndsWith { get; }
+
+		public NamespaceTransform (string oldValue, string newValue)
+		{
+			OldValue = oldValue;
+			NewValue = newValue;
+
+			if (OldValue.EndsWith (".", StringComparison.Ordinal)) {
+				IsStartsWith = true;
+				OldValue = OldValue.Substring (0, OldValue.Length - 1);
+			}
+
+			if (OldValue.StartsWith (".", StringComparison.Ordinal)) {
+				IsEndsWith = true;
+				OldValue = OldValue.Substring (1);
+			}
+		}
+
+		public string ApplyInternal (string value)
+		{
+			string result;
+
+			while (true) {
+				result = ApplyInternal (value);
+
+				if (result == value)
+					return result;
+
+				value = result;
+			}
+		}
+
+		public string Apply (string value)
+		{
+			// Handle a "starts with" transform
+			if (IsStartsWith) {
+				if (value.StartsWith (OldValue, StringComparison.OrdinalIgnoreCase))
+					return (NewValue + value.Substring (OldValue.Length)).TrimStart ('.');
```

Comment:
The *problem* with `source=".Foo."` is *here*: the constructor doesn't "reject" it, but `Apply()` doesn't attempt to match *both* `IsStartsWith` *and* `IsEndsWith`.  Instead, it checks `IsStartsWith` *first*, and then returns a value.  Which means `<ns-replace source=".foo." …/>` is treated identically to `<ns-replace source=".foo" …/>`.

I can see one of two solutions:

 1. Flat out reject `<ns-replace source=".foo." />`: the `NamespaceTransform` constructor should throw when `IsStartsWith && IsEndsWith`.
 2. Support it.

(2) could be done by checking for it *first*:

```cs
if (IsStartsWith && IsEndsWith) {
    if (value.Equals (OldValue, StringComparison.OrdinalIgnoreCase))
        return NewValue;
}
else if (IsStartsWith) …
else if (IsEndsWith) …
```